aws-rds-cfn-common/src/main/java/software/amazon/rds/common/handler/Timestamp.java

```diff
@@ -0,0 +1,31 @@
+package software.amazon.rds.common.handler;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+@lombok.Getter
+@lombok.Setter
+@lombok.ToString
+@lombok.EqualsAndHashCode
+public class Timestamp {
+
+    public interface Provider {
+        Map<String, Long> timestamps = new HashMap<>();
+
+        default void timestamp(final String label, final Instant instant) {
+            timestamps.put(label, instant.getEpochSecond());
+        }
+
+        default void timestampOnce(final String label, final Instant instant) {
+            timestamps.computeIfAbsent(label, s -> instant.getEpochSecond());
+        }
+
+        default Instant getTimestamp(final String label) {
+            if (timestamps.containsKey(label)) {
+                return Instant.ofEpochSecond(timestamps.get(label));
+            }
+            return null;
+        }
+    }
```

Comment:
Consider the following code:

```java
public class Main {
    interface FooIntf {
        Map<String, String> opts = new HashMap<>();

        default String get(String key) {
            return opts.get(key);
        }

        default void put(String key, String value) {
            opts.put(key, value);
        }
    }

    public static void main(String[] args) {

        class Bar implements FooIntf {
        }

        class Boo implements FooIntf {
        }

        Bar bar = new Bar();
        Boo boo = new Boo();

        bar.put("key", "value");
        System.out.println(boo.get("key"));
    }
}
```

This will output `value`. Let's avoid this trap here.